api/src/main/java/org/apache/iceberg/types/AssignFreshIds.java

```diff
@@ -88,11 +88,16 @@ public Type struct(Types.StructType struct, Iterable<Type> futures) {
     for (int i = 0; i < length; i += 1) {
       Types.NestedField field = fields.get(i);
       Type type = types.next();
-      if (field.isOptional()) {
-        newFields.add(Types.NestedField.optional(newIds.get(i), field.name(), type, field.doc()));
-      } else {
-        newFields.add(Types.NestedField.required(newIds.get(i), field.name(), type, field.doc()));
-      }
+      newFields.add(
```

Comment:
can you please add some tests to `TestTypeUtil` that use `TypeUtil.assignFreshIds()`and verify that the defaults are properly propagated?
We might also have the same issue with `TypeUtil.reassignDocs()`, so I think it would be good to add some tests for that as well

Reply:
Good catch!
Fixed `AssignFreshIds`, `AssignIds` and `ReassignDoc` in the end. (`ReassignIds` were working)

Reply:
Had to rebase because of a concurrent commit